pkg/plugins/manager/client/client.go

```diff
@@ -100,12 +108,21 @@ func (s *Service) CallResource(ctx context.Context, req *backend.CallResourceReq
 		removeConnectionHeaders(req.Headers)
 		removeHopByHopHeaders(req.Headers)
 
+		processedStreams := 0
 		wrappedSender := callResourceResponseSenderFunc(func(res *backend.CallResourceResponse) error {
-			if res != nil && len(res.Headers) > 0 {
-				removeConnectionHeaders(res.Headers)
-				removeHopByHopHeaders(res.Headers)
+			// Expected that headers and status are only part of first stream
+			if processedStreams == 0 && res != nil {
+				if len(res.Headers) > 0 {
+					removeConnectionHeaders(res.Headers)
+					removeHopByHopHeaders(res.Headers)
+					removeNonAllowedHeaders(res.Headers)
```

Comment:
Given these changes and the dependency on `proxyutil`, this feels more like `client.go` is a Grafana-specific implementation of `plugins.Client`. Maybe it belongs in `pluginsintegration` eventually and we have a "simple" plugin client implementation inside `pkg/plugins` that we wrap. WDYT?

Reply:
Yeah I did think about that. Thought temporary solution until we move proxy downwards, but I think you're right. Using a client middleware for this probably make more sense :+1:

Reply:
Fixed. Moved proxyutil call to middleware and left removal of set-cookie